### PR TITLE
use :maps.with and :maps.without for Map.take and Map.drop respectively

### PR DESCRIPTION
**Pull request by benwilson512**
Can't see why not, they're bound to be faster too. I hope the inlining is ok, seemed reasonable to me.

### CODE REVIEW

**Comment by benwilson512** [comment]
Turns out this is a Grade A bad idea :) Initial benchmarking is showing that our current implementations are faster.

Upon further look at the implementation, this makes sense. The erlang one rebuilds the entire map: https://github.com/erlang/otp/blob/maint/lib/stdlib/src/maps.erl#L208. This seems... wrong

File: lib/elixir/lib/map.ex [hunk 1]
```diff
@@ -10,7 +10,7 @@ defmodule Map do
 
   @type key :: any
   @type value :: any
-  @compile {:inline, fetch: 2, put: 3, delete: 2, has_key?: 2}
+  @compile {:inline, fetch: 2, put: 3, delete: 2, has_key?: 2, drop: 2, take: 2}
```
Comment (ericmj) [inline lib/elixir/lib/map.ex]: Inlining only works inside the current module so this wouldn't do anything since drop and take are not used here.

**Comment by ericmj** [comment]
Thanks for bringing this up with OTP.

File: lib/elixir/lib/map.ex (hunk 1 above)
Comment (benwilson512) [inline lib/elixir/lib/map.ex]: Ah good to know thanks.

**Comment by benwilson512** [comment]
My thought is that any improvements we think up with the OTP team (like your Map.from_list approach to Map.take) we'll pull over after the erlang side is settled in another PR.

**Comment by psyeugenic** [comment]
I would like to add my thanks for bringing up this to us at OTP!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request closed without merging**